cpa/tester/policies/helpers/contexts/policy_test.yaml

```diff
@@ -0,0 +1,170 @@
+test_allowlist:
+  meta:
+    project_id: a
+    project_list: a
+    allowlist: [c1, c2]
+  input:
+    workflows:
+      main:
+        jobs:
+          - test:
+              context: c1
+          - build:
+              context: c2
+  decision: &root_decision
+    status: PASS
+    enabled_rules:
+      - context_allowlist
+      - context_blocklist
+      - context_branch_allowlist
+
+  cases:
+    invalid_context:
+      meta:
+        allowlist: [c0, c1]
+      decision:
+        <<: *root_decision
+        status: SOFT_FAIL
+        soft_failures:
+          - rule: context_allowlist
+            reason: 'main.build: uses context value(s) not in allowlist for project: c2'
+    invalid_contexts:
+      meta:
+        allowlist: [v1, v2]
+      decision:
+        <<: *root_decision
+        status: SOFT_FAIL
+        soft_failures:
+          - rule: context_allowlist
+            reason: 'main.build: uses context value(s) not in allowlist for project: c2'
+          - rule: context_allowlist
+            reason: 'main.test: uses context value(s) not in allowlist for project: c1'
+
+    test_unaffected_project:
+      meta:
+        project_list: [a, b, c]
+        project_id: x
+        allowlist: [v1, v2]
+      decision: *root_decision
+
+    test_multiple_invalid_contexts_in_job:
+      input:
+        workflows:
+          main:
+            jobs:
+              - test:
+                  context: [bad, dangerous, secret]
+      decision:
+        <<: *root_decision
+        status: SOFT_FAIL
+        soft_failures:
+          - rule: context_allowlist
+            reason: 'main.test: uses context value(s) not in allowlist for project: bad, dangerous, secret'
+
+test_blocklist:
+  meta:
+    project_list: a
+    project_id: a
+    blocklist: [bad, dangerous, secret]
```

Comment:
the others are simply named allowlist, and since this is the tests in the contexts directory, I don't feel like getting very explicit with naming adds enough value versus all the other changes this one we cascade.

done the others but this one I think will stay as is!